modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -317,6 +317,7 @@ void GifDecoder::code2pixel(Mat& img, int start, int k){
 bool GifDecoder::lzwDecode() {
     // initialization
     lzwMinCodeSize = m_strm.getByte();
+    if (lzwMinCodeSize > 12) return false;
```

Comment:
`int lzwCodeSize = lzwMinCodeSize + 1;`
Why the following `CV_Assert(lzwCodeSize > 2 && lzwCodeSize <= 12);` is not enough?

Reply:
I propose to use CV_Assert, if the value is not expected in correct gif.

Reply:
Ah indeed, thx, I missed that. I am moving it above the << operation which triggered the fuzzer